express.js

```diff
@@ -269,372 +336,877 @@ var resizeImage = (size, double) => {
-		res.locals.port = process.env.PORT || 8118
-		next()
+	let msg = ''
+	let error = false
+	try {
+		await fs.promises.unlink(path.join(qsets, "log.json"), (err) => {
+			if (err) {
+				res.status(204)
+				msg = "log.json not found"
+			} else {
+				res.status(200);
+				msg = "removed log.json"
+			}
+		});
+	} catch(err) {
+		msg = "log.json not found"
+		error = true
+		res.status(204)
+	}
+	hasPlayLogs = false;
+	return res.json({ error: error, msg: msg})
+})
+
+// Preview widget scores
+app.get('/mwdk/scores/preview/:id?', (req, res) => {
+	res.locals = Object.assign(res.locals, { template: 'score_mwdk', IS_PREVIEW: 'true'})
+	res.render(res.locals.template)
+})
+// Play widget scores
+app.get(['/mwdk/scores/:id?'], (req, res) => {
+	res.locals = Object.assign(res.locals, { template: 'score_mwdk', IS_PREVIEW: 'false'})
+	res.render(res.locals.template)
+})
+
+// The create page frame that loads the widget creator
+// Must have hash '1' to work
+app.get('/mwdk/widgets/1-mwdk/create', (req, res) => {
+	res.locals = Object.assign(res.locals, {template: 'creator_mwdk', instance: req.params.hash || '1'})
+	res.render(res.locals.template, { layout: false})
+});
+
+app.get('/mwdk/widgets/1-mwdk/creators-guide', (req, res) => {
+	res.locals = Object.assign(res.locals, {
+		template: 'guide_page',
+		name: '1-mwdk',
+		type: 'creator',
+		hasPlayerGuide: true,
+		hasCreatorGuide: true,
+		docPath: '/guides/creator.html',
+		instance: req.params.hash || '1'
 	})
+	res.render(res.locals.template, { layout: false})
+})
+
+app.get('/mwdk/widgets/1-mwdk/players-guide', (req, res) => {
+	res.locals = Object.assign(res.locals, {
+		template: 'guide_page',
+		name: '1-mwdk',
+		type: 'player',
+		hasPlayerGuide: true,
+		hasCreatorGuide: true,
+		docPath: '/guides/player.html',
+		instance: req.params.hash || '1'
+	})
+	res.render(res.locals.template, { layout: false})
+})
+
+// Show the package options
+app.get('/mwdk/package', (req, res) => {
+	res.locals = Object.assign(res.locals, {template: 'download'})
+	res.render(res.locals.template)
+})
+
+// Build and download the widget file
+app.get('/mwdk/download', (req, res) => {
+	let { widgetPath, widgetData } = buildWidget()
+	res.set('Content-Disposition', `attachment; filename=${widgetData.clean_name}.wigt`);
+	res.send(fs.readFileSync(widgetPath));
+});
+
+
+app.get('/mwdk/helper/annotations', (req, res) => {
+	res.locals = Object.assign(res.locals, {template: 'helper-annotator', title: 'annotate yo widget'})
+	res.render(res.locals.template)
+});
+
+app.get('/mwdk/install', (req, res) => {
+	res.write('<html><body id="result"><pre>');
+	// Find the docker-compose container for materia-web
+	// 1. lists all containers
+	// 2. filter for materia-web image and named xxxx_phpfpm_1 name
+	// 3. pick the first line
+	// 4. pick the container name
+	let targetImage = execSync('docker ps -a --format "{{.Image}} {{.Names}}" | grep -e ".*materia:.* docker_app_.*" | head -n 1 | cut -d" " -f2');
```

Comment:
This line continues to give us problems. Some versions of docker-compose use hyphens, others use underscores, and the server does not handle a search failure well. We can modify the search pattern grep is using to: `".*materia:.* docker[-_]app[-_].*"` which will catch either option.

Reply:
Also it'd be nice if the server gave a more well-handled response if the docker install process runs into an issue